common/hitsystem.go

```diff
@@ -0,0 +1,155 @@
+package common
+
+import (
+	"engo.io/ecs"
+	"engo.io/engo"
+)
+
+type HitGroup byte
+
+//HitBox is currently a simple bounds rect.
+//I hope to add a possibility of internal rects, for more fine grain hit tests at some point
+type HitBox struct {
```

Comment:
Couldn't we use an engo.AABB? There's also an engo.Container interface that I think would be a good fit for this.